blog/2024-01-24-pants-2-19/index.mdx

```diff
@@ -0,0 +1,79 @@
+
+Keep reading to see more, or check the [full changelog](https://github.com/pantsbuild/pants/tree/main/src/python/pants/notes/2.19.x.md). Pants is an open-source project, and the below changes are all contributed by our community. If you want to see something more in the next changelog, join us on [GitHub](https://github.com/pantsbuild/pants) and become a contributor.
+
+We also offer [formal sponsorship tiers for companies](https://www.pantsbuild.org/sponsorship), as well as individual sponsorships via [GitHub](https://github.com/sponsors/pantsbuild). These help pay for the ongoing development and hosting costs, and are managed by the Pants Build non-profit organization.
+
+### New backends and subsystems
+
+In this release, we welcome two new backends and one new subsystem.
+
+[The new `openapi_format` backend](https://www.pantsbuild.org/2.19/reference/subsystems/openapi-format) supports running [openapi-format](https://github.com/thim81/openapi-format) to format `openapi_...` targets. You can enable this by adding `pants.backend.experimental.openapi.lint.openapi_format` to the list of active backends.
+
+[The new `semgrep` backend](https://www.pantsbuild.org/2.19/reference/subsystems/semgrep) supports running [Semgrep](https://semgrep.dev) as a local linter on all files that Pants knows about (any target with a `source` or `sources` field). Configuration files are automatically discovered. You can enable this by adding `pants.backend.experimental.tools.semgrep` to the list of active backends.
+
+[The new `[system-binaries]` subsystem](https://www.pantsbuild.org/2.19/reference/subsystems/system-binaries) allows controlling where Pants searches for binaries like `bash`, `zip`, `tar` that is uses as part of normal operation. Set [the `system_binaries_paths`](https://www.pantsbuild.org/2.19/reference/subsystems/system-binaries#system_binary_paths) as appropriate, potentially using the `<PATH>` special value. This can also be set on specific [environments](https://www.pantsbuild.org/2.19/docs/using-pants/environments) using the `system_binaries_system_binary_paths` field.
+
+### Built-in test retries (for Python)
+
+Python tests can now be [attempted multiple times](https://www.pantsbuild.org/2.19/docs/using-pants/using-pants-in-ci#tip-automatically-retry-failed-tests), if they fail. This can help with limiting the impact of flaky tests. For instance, to require tests to fail three times before the overall `pants test` goal fails, set [`[test].attempts_default = 3`](https://www.pantsbuild.org/2.19/reference/goals/test#attempts_default). In this case, a very flaky test might fail on the first and second runs but finally pass on the third run, and thus the overall Pants invocation would pass.
+
+This functionality can be added to any other language with fairly little work. Do you want automatic retries for Scala, Go, or another language Pants supports? Come chat to us in [#development with Slack](https://pantsbuild.slack.com).
```

Comment:
Potentially this link should be the CTA Google form link? And/or maybe we could link to https://github.com/pantsbuild/pants/issues/20378 as the central coordination.

Reply:
Yeah, I asked about the Google Forms link in the maintainers channel on Slack, will change it to there. Felt weird to link to link a Google Forms as a CTA, but will do so. And will add the issue too, thanks!